File: task/src/main/java/com/dh/task/data/source/local/TasksLocalDataSource.java
package com.dh.task.data.source.local;

import androidx.annotation.NonNull;

import com.dh.core.callback.DataCallback;
import com.dh.task.data.Task;
import com.dh.task.data.source.TasksDataSource;

import java.util.List;

import javax.inject.Inject;

/**
 * Created by Jin on 2020/10/26.
 * Description 本地数据源
 */
public class TasksLocalDataSource implements TasksDataSource {

    @Inject
    public TasksLocalDataSource() {

    }

    @Override
    public void getTasks(@NonNull DataCallback<List<Task>> callback) {
        callback.onDataNotAvailable();
    }

    @Override
    public void saveTask(@NonNull Task task) {

    }
}

File: task/src/main/java/com/dh/task/data/source/local/TasksLocalDataSource.kt
package com.dh.task.data.source.local

import com.dh.core.callback.DataCallback
import com.dh.task.data.Task
import com.dh.task.data.source.TasksDataSource
import javax.inject.Inject

/**
 * Created by Jin on 2021/2/24.
 * Description 本地数据源
 */
class TasksLocalDataSource @Inject constructor() : TasksDataSource {

    override fun getTasks(callback: DataCallback<MutableList<Task>?>) {
        callback.onDataNotAvailable()
    }

    override fun saveTask(task: Task) {
    }
}